test/decomposer/test_aba_decomposer.py

```diff
@@ -31,15 +32,17 @@ def test_specific_bloch_rotation(aba_decomposer: Callable[..., Decomposer]) -> N
 @pytest.mark.parametrize("aba_decomposer", ABA_DECOMPOSER_LIST)
 def test_all_octants_of_bloch_sphere_rotation(aba_decomposer: Callable[..., Decomposer]) -> None:
     decomposer = aba_decomposer()
-    steps = 6
+    steps = 5
     phase_steps = 3
     coordinates = np.linspace(-1, 1, num=steps)
     angles = np.linspace(-2 * np.pi, 2 * np.pi, num=steps)
     phases = np.linspace(-np.pi, np.pi, num=phase_steps)
     axes = [[i, j, z] for i in coordinates for j in coordinates for z in coordinates]
 
-    for angle in angles:
-        for axis in axes:
+    for axis in axes:
+        if all(abs(component) < ATOL for component in axis):  # not a valid axis
+            continue
```

Comment:
Good solution so that we can now include 0 as a value for the axis components (i.e. 5 steps).

Reply:
I think so.